### PR TITLE
feat(git): add base types for aws codecommit 

### PR DESCRIPTION
**Pull request by overbit**
Partial: #6351 

### Summary
🚀🧪🧹

The pull request adds support for the AWS CodeCommit provider as a possible option for the git provider. It implements the `GitFactory` and the `AwsCodeCommitService` classes, and defines the types and the validation function for the AWS CodeCommit provider organization properties. It also renames some files and classes, updates some imports and exports, increases the coverage threshold, and adds the unit tests for the `AwsCodeCommitService` class. It modifies the `package.json` file to add the dependency of `@aws-sdk/client-codecommit`.

> _`AwsCodeCommitService`_
> _New class for git provider_
> _Winter of testing_

### Walkthrough
*  Rename `GitService` class to `GitClientService` and update import and export statements ([link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-a1600a4494f1ff70306b12cca22bd3001ad469f8a0803727550a74e4126b1c3aL1-R1),[link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-092916068d0197966c65911f85bb41cea4ebbf6c39c9f971a5f953b7e93f509aL1-R1),[link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-51391ca3053c92c35199312b6ec54cbfb6f75cd774a6d36aa1b938e7625c6b1cL13-R14))
*  Add `AwsCodeCommit` value to `EnumGitProvider` enum and define `AwsCodeCommitProviderOrganizationProperties` interface and validation function in `types.ts` file ([link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-32aa29da64da9b56bdccbd8c64390608dd92173a29efa2ccdd76128047fe4eb0R16),[link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-32aa29da64da9b56bdccbd8c64390608dd92173a29efa2ccdd76128047fe4eb0R88-R118))
*  Create `AwsCodeCommitService` class that implements `GitProvider` interface and provide constructor and basic validation in `aws-code-commit.service.ts` file ([link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-79b256dbe34679386e43a1af887c0d3a77531fcf3f7b69683a0dffe258c7262aR1-R144))
*  Add unit tests for `AwsCodeCommitService` class in `aws-code-commit.service.spec.ts` file ([link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-0187c463cbbf3cb74e5c005c9691d0584432bec13114b641714134e483c23486R1-R184))
*  Add case for `EnumGitProvider.AwsCodeCommit` in `createGitProvider` method of `GitFactory` class and update imports in `git-factory.ts` file ([link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7L5-R5),[link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7R12-R13),[link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7L18),[link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7R49-R63))
*  Add dependency of `@aws-sdk/client-codecommit` to `package.json` file ([link](https://github.com/amplication/amplication/pull/6427/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R36))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: libs/util/git/src/providers/aws/aws-code-commit.service.spec.ts
Comment (abrl91) [inline libs/util/git/src/providers/aws/aws-code-commit.service.spec.ts:1]: 😲

File: libs/util/git/jest.config.ts [hunk 1]
```diff
@@ -2,16 +2,21 @@
...
   },
-  moduleFileExtensions: ["ts", "tsx", "js", "jsx"],
+  moduleFileExtensions: ["ts", "js", "html"],
   coverageDirectory: "../../../coverage/libs/util/git",
   coverageThreshold: {
     global: {
-      branches: 31.25,
-      lines: 25,
+      branches: 80,
+      lines: 40,
```
Comment (yuval-hazaz) [inline libs/util/git/jest.config.ts:19]: Should this be part of this PR?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,6 +35,8 @@ jobs:
   ci:
     name: Continuous Integration
     runs-on: ubuntu-20.04
+    env:
+      NODE_OPTIONS: --max_old_space_size=4096
```
Comment (yuval-hazaz) [inline .github/workflows/ci.yml:39]: Should this be part of this PR?

**Review by yuval-hazaz — APPROVED** [review]
couple of comments about unexpected changes
The code changes look good

File: libs/util/git/jest.config.ts (hunk 1 above)
Comment (overbit) [reply libs/util/git/jest.config.ts:19]: This was done due to the new test added as part of this PR

File: .github/workflows/ci.yml (hunk 2 above)
Comment (overbit) [reply .github/workflows/ci.yml:39]: Reverted

**Pull request merged**